Sources/Data Model/ProjectConfig.swift

```diff
@@ -122,6 +124,41 @@ class ProjectConfig {
             project.rollouts.forEach { map[$0.id] = $0 }
             return map
         }()
+        
+        self.flagRulesMap = {
+            var map = [String: [Experiment]]()
+            
+            project.featureFlags.forEach { flag in
+                var experiments = flag.experimentIds.compactMap { expId in
+                    return allExperiments.filter { $0.id == expId }.first
+                }
+                
+                let rollout = project.rollouts.filter { $0.id == flag.rolloutId }.first
+                experiments.append(contentsOf: rollout?.experiments ?? [])
+                
+                map[flag.key] = experiments
+            }
+            
+            return map
+        }()
+        
+        self.flagVariationsMap = {
+            var map = [String: [Variation]]()
+            
+            flagRulesMap.forEach { (flagKey, rules) in
+                var variations = [Variation]()
+                rules.forEach { rule in
+                    rule.variations.forEach { variation in
+                        if !variations.map({ $0.id }).contains(variation.id) {
```

Comment:
why this condition is needed? don't we have unique variations inside rollouts and experiments?

Reply:
We need "all variations for flag" to check if forced-decision is valid and to get the forced variation from its key. Since we do not have such "all variations for flag" in datafile, this code tried to build "variations" for a flag by walking through variations of all experiement/delivery rules. The condition is to avoid redundant listing.

Reply:
Added comments